kpatch-build/create-diff-object.c

```diff
@@ -1923,6 +1923,28 @@ static void kpatch_regenerate_special_section(struct kpatch_elf *kelf,
 	sec->base->data->d_size = dest_offset;
 }
 
+static void kpatch_check_relocations(struct kpatch_elf *kelf)
+{
+	struct rela *rela, *safe;
+	struct section *sec;
+	Elf_Data *sdata;
+
+	list_for_each_entry(sec, &kelf->sections, list) {
+		if (!is_rela_section(sec))
+			continue;
+		list_for_each_entry_safe(rela, safe, &sec->relas, list) {
+			if (rela->sym->sec) {
+				sdata = rela->sym->sec->data;
+				if (rela->addend < sdata->d_off ||
```

Comment:
On x86, rela->addend is often negative in relation to the symbol, e.g.:
```
Relocation section [20] '.rela.text.semctl_stat' for section [19] '.text.semctl_stat' at offset 0x720 contains 6 entries:
  Offset              Type            Value               Addend Name
  0x0000000000000001  X86_64_NONE     000000000000000000      -4 __fentry__
  0x000000000000002d  X86_64_PC32     000000000000000000      -4 ipc_obtain_object_check
  0x000000000000004f  X86_64_PC32     000000000000000000      -4 ipcperms
  0x0000000000000062  X86_64_PC32     000000000000000000      -4 security_sem_semctl
  0x0000000000000074  X86_64_PC32     000000000000000000      -4 ipc_obtain_object_idr
  0x0000000000000090  X86_64_PC32     000000000000000000      -4 kernel_to_ipc64_perm
```
That's because on x86, the relative address is adjusted by the length of the instruction which uses it.

So for X86_64_PC32 relocations, the check may need to take into account the length of the instruction which uses the rela's address.  Though that's probably overkill.  Instead I would recommend just removing the `<` check, since I don't think we don't change the addend anyway and I don't see how it could go beyond the beginning of the section.

The `>` check is much more important IMO.